autoscheduler/scraper/tests/api_tests.py

```diff
@@ -0,0 +1,172 @@
+        """ Tests that /api/course?dept=CSCE&course_num=181&term=201931 gives the
+            correct output
+        """
+        # Arrange
+        expected = CourseSerializer(self.courses[0])
+        data = {'dept': 'CSCE', 'course_num': '181', 'term': '201931'}
+
+        # Act
+        response = self.client.get("/api/course", data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()[0], expected.data)
+
+    def test_api_course_gives_valid_response_law(self):
+        """ Tests that /api/course?dept=LAW&course_num=7500S&term=202031 gives the
+            correct output (verifies API can handle null credit_hours)
+        """
+        # Arrange
+        expected = CourseSerializer(self.courses[4])
+        data = {'dept': 'LAW', 'course_num': '7500S', 'term': '202031'}
+
+        # Act
+        response = self.client.get("/api/course", data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()[0], expected.data)
+
+    def test_api_section_serializer_gives_expected_output(self):
+        """ Tests that the section serializer yields the correct data """
+        # Arrange
+        first_start = time(11, 30)
+        first_end = time(12, 20)
+        second_start = time(9, 10)
+        second_end = time(10)
+        expected = {
+            'instructor_gpa': None,
+            'honors_only': False,
+            'web_only': False,
+            'meeting_times': [[first_start, first_end], [second_start, second_end]]
+        }
+
+        # Act
+        serializer = SectionSerializer(self.sections[0])
+
+        # Assert
+        self.assertEqual(expected, serializer.data)
+
+    def test_api_section_gives_valid_response(self):
+        """ Tests that /api/section?id=000001&term=201931 gives the correct output """
+        # Arrange
+        expected = SectionSerializer(self.sections[0])
+        data = {'id': '000001', 'term': '201931'}
+
+        # Act
+        response = self.client.get("/api/section", data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()[0], expected.data)
+
+    def test_api_course_search_gives_correct_results_cs(self):
+        """ Tests that /api/course/search?search=CS&term=201931 gives correct output """
+        # Arrange
+        expected = {'results': ["CSCE 181", "CSCE 315"]}
+        data = {'search': 'CS', 'term': '201931'}
+
+        # Act
+        response = self.client.get("/api/course/search", data=data)
+
+        # Assert
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()[0], expected)
+
+    def test_api_course_search_gives_correct_results_c(self):
+        """ Tests that /api/course/search?search=C&term=201931 gives correct output """
+        # Arrange
+        expected = {'results': ["COMM 203", "CSCE 181", "CSCE 315"]}
```

Comment:
Why is COMM 203 included if the search query is `'CS'`?

Reply:
Good point, I meant to make the query `'C'` (like the function name and docstring indicate) but copy/pasted the test case for `'CS'` and forgot to change the search term in `data`